tools/devops/provision-xcode.csx

```diff
@@ -0,0 +1,90 @@
+using System.Collections.Generic;
+using System.IO;
+using System.Net;
+using System.Linq;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+
+using Newtonsoft.Json.Linq;
+
+using Xamarin.Provisioning;
+using Xamarin.Provisioning.Model;
+
+// Provision Xcode
+//
+// Overrides:
+// * The current commit can be overridden by setting the PROVISION_FROM_COMMIT variable.
+
+var commit = Environment.GetEnvironmentVariable ("BUILD_SOURCEVERSION");
+var provision_from_commit = Environment.GetEnvironmentVariable ("PROVISION_FROM_COMMIT") ?? commit;
+
+// Looks for a variable either in the environment, or in current repo's Make.config.
+// Returns null if the variable couldn't be found.
+IEnumerable<string> make_config = null;
+string FindConfigurationVariable (string variable, string hash = "HEAD")
+{
+	var value = Environment.GetEnvironmentVariable (variable);
+	if (!string.IsNullOrEmpty (value))
+		return value;
+
+	if (make_config == null)
+		make_config = Exec ("git", "show", $"{hash}:Make.config");
+	foreach (var line in make_config) {
+		if (line.StartsWith (variable + "=", StringComparison.Ordinal))
+			return line.Substring (variable.Length + 1);
+	}
+
+	return null;
+}
+
+string FindVariable (string variable)
+{
+	var value = FindConfigurationVariable (variable, provision_from_commit);
+	if (!string.IsNullOrEmpty (value))
+		return value;
+
+	throw new Exception ($"Could not find {variable} in environment nor in the commit's ({commit}) manifest.");
+}
+
+void ExecVerbose (string filename, params string[] args)
+{
+	Console.WriteLine ($"{filename} {string.Join (" ", args)}");
+	Exec (filename, args);
+}
+
+void ListXcodes ()
+{
+	Console.WriteLine ($"Xcodes:");
+	var lines = Exec ("bash", "-c", "ls -lad /Applications/Xcode*");
```

Comment:
fragile but it will work for us (by convention)
if you right click on files you get a list of all installed Xcode (likely safer, but I'm not sure how it's exposed, Finder?)

Reply:
Right click where exactly?

Reply:
e.g. right click on "Entitlements.plist", select "Open With" and you see if current (selected) Xcode first then a list of all installed Xcode versions/apps (and maybe more apps configured to edit plist)